File: android/api-tests/src/test/java/com/revenuecat/apitests/java/OnResultApiTests.java
package com.revenuecat.apitests.java;

import com.revenuecat.purchases.hybridcommon.ErrorContainer;
import com.revenuecat.purchases.hybridcommon.OnResult;

import java.util.Map;

@SuppressWarnings("unused")
class OnResultApiTests {
    private void checkOnSuccess(OnResult onResult,
                                Map<String, Object> result) {
        onResult.onReceived(result);
    }

    private void checkOnError(OnResult onResult,
                              ErrorContainer errorContainer) {
        onResult.onError(errorContainer);
    }
}

File: android/api-tests/src/test/java/com/revenuecat/apitests/kotlin/OnResultApiTests.kt
package com.revenuecat.apitests.kotlin

import com.revenuecat.purchases.hybridcommon.ErrorContainer
import com.revenuecat.purchases.hybridcommon.OnResult

@Suppress("unused")
private class OnResultApiTests {
    fun checkOnReceived(
        onResult: OnResult,
        resultMap: Map<String, Any>,
    ) {
        onResult.onReceived(resultMap)
    }

    fun checkOnError(
        onResult: OnResult,
        errorContainer: ErrorContainer,
    ) {
        onResult.onError(errorContainer)
    }
}
